boards: adi: Add AD-APARD32690-SL

CiprianRegus opened the pull request:
Add board support for the ADI AD-APARD32690-SL. The platform is based on a MAX32690 microcontroller.

Tested with the following samples:

* blinky
* button
* sockets/echo

The board's product page may be accessed here: https://www.analog.com/en/resources/evaluation-hardware-and-software/evaluation-boards-kits/ad-apard32690-sl.html

CiprianRegus committed:
boards: adi: Add AD-APARD32690-SL

Add board support for the ADI AD-APARD32690-SL. The platform is based on
a MAX32690 microcontroller, it has an Arduino-Mega compatible header and
2 PMOD connectors. Also, a 10 Mbps single-pair Ethernet link is enabled
by using the ADIN1110 10BASE-T1L MAC/PHY.

Signed-off-by: Ciprian Regus <ciprian.regus@analog.com>

galak reviewed (dismissed):
LGTM.

dleach02 commented on boards/adi/apard32690/doc/img/apard32690_img1.jpg:
There appears to be a push to have images use webp format. I recommend converting both images.

dleach02 commented:
other then the image it LGTM

CiprianRegus commented:
V3:
- Converted the images to the webp format.

microbuilder commented:
@CiprianRegus Can you have a look at the build errors in CI to hopefully get this in before freeze this Friday? Thanks!

MaureenHelm commented on boards/adi/apard32690/apard32690_max32690_m4.dts:
```diff
@@ -0,0 +1,160 @@
+	status = "okay";
+};
+
+&uart3 {
+	status = "okay";
+	current-speed = <1000000>;
+	data-bits = <8>;
+};
```
No pinctrl for this node? I also see that this node is chosen for `zephyr,bt-uart`. Have you tested this?

CiprianRegus replied:
I have not tested the Bluetooth functionality. The `zephyr,bt-uart` property was only added since the CI  would fail to compile `samples/bluetooth/peripheral_hr` otherwise. Is there a better fix for this?

MaureenHelm replied:
`zephyr,bt-uart` gets set in `boards/shields/frdm_kw41z/frdm_kw41z.overlay` so you shouldn't need to set it here. You should be able to fix the CI failure by cherry-picking #74038. You'll then also be able to remove the `uart3` node which I presume you're using as a placeholder just to fake out CI.

MaureenHelm commented on boards/adi/apard32690/apard32690_max32690_m4.dts:
```diff
@@ -0,0 +1,160 @@
+	adin1110: adin1110@0 {
+		compatible = "adi,adin1110";
+		reg = <0x0>;
+		spi-max-frequency = <25000000>;
```
```suggestion
		spi-max-frequency = <DT_FREQ_M(25)>;
```

MaureenHelm reviewed (commented):
Need to address the failed compliance check or rebase after merging #74007

MaureenHelm committed:
drivers: serial: Set default baud rate and data bits in max32 driver

The uart-controller devicetree binding declares current-speed and
data-bits properties as optional, but the max32 uart driver failed to
build if they weren't defined. Fix the driver by falling back to
commonly used values for these properties.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

CiprianRegus commented:
V5:
- Cherry-picked https://github.com/zephyrproject-rtos/zephyr/pull/73910/commits/80d5e2e7b8a8bfca3eb3a04d8effdef1d5f91de6 in order to fix CI errors, so https://github.com/zephyrproject-rtos/zephyr/pull/74038 has to be merged first.
- Removed the `zephyr,bt-uart` property from the `chosen` node.
- Used `DT_FREQ_M()` in order to set `spi-max-frequency`.

kartben commented on boards/adi/apard32690/doc/index.rst:
```diff
@@ -0,0 +1,220 @@
+   :align: center
+   :alt: AD-APARD32690-SL Front
+
+.. image:: img/apard32690_img1.webp
+   :align: center
+   :alt: AD-APARD32690-SL Back
+
```
there's a typo (img1 instead of img2) but I would actually ask that you drop the back image entirely, There's no real value justifying adding 50K to the main git repo. In fact, the fact that no one noticed during review kind of confirms my point, probably :) 

![image](https://github.com/zephyrproject-rtos/zephyr/assets/128251/42dd0c67-d276-416a-a9e2-96576bcc6d75)

kartben replied:
On a related note, there seems to be better quality images (better lighting at least) available from offical web page https://www.analog.com/en/resources/evaluation-hardware-and-software/evaluation-boards-kits/ad-apard32690-sl.html (almost feels like the ones you used might be from an early prototype? :) )

see e.g. 

![EVAL-MAX32690-ARDZ_ANGLE-web](https://github.com/zephyrproject-rtos/zephyr/assets/128251/4676e790-c179-4bee-82d5-d437011a1c82)

or

![EVAL-MAX32690-ARDZ_TOP-web](https://github.com/zephyrproject-rtos/zephyr/assets/128251/11d3a476-5ee0-48d3-bd60-210795b10f75)

CiprianRegus replied:
Hi @kartben, and thank you for pointing this out. The images which I've added as part of this PR are actually of the current revision of the board, so even though they are of lower quality, I think it would be better to use them instead of the ones from official web page (which is actually the earlier prototype you can't buy anymore). I'll drop the back image.

kartben reviewed (approved):
docs is a go - thanks!
